pkg/monitor/intervalcreation/node.go

```diff
@@ -100,3 +114,185 @@ func IntervalsFromEvents_NodeChanges(events monitorapi.Intervals, _ monitorapi.R
+	if !strings.Contains(logLine, `Probe failed`) {
+		return nil
+	}
+	if !strings.Contains(logLine, `probeType="Readiness"`) {
+		return nil
+	}
+
+	failureOutputRegex.MatchString(logLine)
+	if !failureOutputRegex.MatchString(logLine) {
+		return nil
+	}
+	outputSubmatches := failureOutputRegex.FindStringSubmatch(logLine)
+	message := outputSubmatches[1]
+	message, _ = strconv.Unquote(`"` + message + `"`)
+
+	containerRef := probeProblemToContainerReference(logLine)
+	failureTime := kubeletLogTime(logLine)
+	return monitorapi.Intervals{
+		{
+			Condition: monitorapi.Condition{
+				Level:   monitorapi.Info,
+				Locator: containerRef.ToLocator(),
+				Message: monitorapi.ReasonedMessage(monitorapi.ContainerReasonReadinessFailed, message),
+			},
+			From: failureTime,
+			To:   failureTime,
+		},
+	}
+}
+
+func readinessError(logLine string) monitorapi.Intervals {
+	if !strings.Contains(logLine, `Probe errored`) {
+		return nil
+	}
+	if !strings.Contains(logLine, `probeType="Readiness"`) {
+		return nil
+	}
+
+	errorOutputRegex.MatchString(logLine)
+	if !errorOutputRegex.MatchString(logLine) {
+		return nil
+	}
+	outputSubmatches := errorOutputRegex.FindStringSubmatch(logLine)
+	message := outputSubmatches[1]
+	message, _ = strconv.Unquote(`"` + message + `"`)
+
+	containerRef := probeProblemToContainerReference(logLine)
+	failureTime := kubeletLogTime(logLine)
+	return monitorapi.Intervals{
+		{
+			Condition: monitorapi.Condition{
+				Level:   monitorapi.Info,
+				Locator: containerRef.ToLocator(),
+				Message: monitorapi.ReasonedMessage(monitorapi.ContainerReasonReadinessErrored, message),
+			},
+			From: failureTime,
+			To:   failureTime,
+		},
+	}
+}
+
+var containerRefRegex = regexp.MustCompile(`pod="(?P<NS>[a-z0-9.-]+)\/(?P<POD>[a-z0-9.-]+)" podUID=(?P<PODUID>[a-z0-9.-]+) containerName="(?P<CONTAINER>[a-z0-9.-]+)"`)
+var failureOutputRegex = regexp.MustCompile(`"Probe failed" probeType="Readiness".*output="(?P<OUTPUT>.+)"`)
+var errorOutputRegex = regexp.MustCompile(`"Probe errored" err="(?P<OUTPUT>.+)" probeType="Readiness"`)
+
+func probeProblemToContainerReference(logLine string) monitorapi.ContainerReference {
+	ret := monitorapi.ContainerReference{}
+	containerRefRegex.MatchString(logLine)
+	if !containerRefRegex.MatchString(logLine) {
+		return ret
+	}
+
+	subMatches := containerRefRegex.FindStringSubmatch(logLine)
+	subNames := containerRefRegex.SubexpNames()
+	for i, name := range subNames {
+		switch name {
+		case "NS":
+			ret.Pod.Namespace = subMatches[i]
+		case "POD":
+			ret.Pod.Name = subMatches[i]
+		case "PODUID":
+			ret.Pod.UID = subMatches[i]
+		case "CONTAINER":
+			ret.ContainerName = subMatches[i]
+		}
+	}
+
+	return ret
+}
+
+var kubeletTimeRegex = regexp.MustCompile(`^(?P<MONTH>\S+)\s(?P<DAY>\S+)\s(?P<TIME>\S+)`)
+
+func kubeletLogTime(logLine string) time.Time {
+	kubeletTimeRegex.MatchString(logLine)
+	if !kubeletTimeRegex.MatchString(logLine) {
+		return time.Now()
```

Comment:
Per below this confuses me, why would we assume time.Now(), wouldn't that mess up intervals chart?

Reply:
> Per below this confuses me, why would we assume time.Now(), wouldn't that mess up intervals chart?

what value would you prefer?